autoendpoint/src/server/extractors/subscription.rs

```diff
@@ -116,9 +122,15 @@ fn extract_public_key(vapid: &VapidHeader, token_info: &TokenInfo) -> ApiResult<
                 )
             })?;
 
-            public_key.to_string()
+            VapidHeaderWithKey {
+                vapid,
+                public_key: public_key.to_string(),
+            }
         }
-        VapidVersionData::Version2 { public_key } => public_key.clone(),
+        VapidVersionData::Version2 { public_key } => VapidHeaderWithKey {
+            public_key: public_key.clone(),
+            vapid,
```

Comment:
Property order can be fixed. The reason it was different is because `public_key` is `&String`, a reference to `vapid.version_data`. We can't move `vapid` into `VapidHeaderWithKey` until we stop borrowing `public_key` (via the call to clone). This is fixed by cloning `vapid.version_data` at the beginning of the match block.

Clippy didn't say anything about cloning `public_key` since it is a `&String` (it has to be cloned to get a `String`). A side effect of cloning `vapid.version_data` at the beginning of the match is that we don't need to clone `public_key` here anymore.